packages/framework/ai-collab/src/explicit-strategy/agentEditReducer.ts

```diff
@@ -184,8 +196,29 @@ export function applyAgentEdit(
 			let insertedObject: TreeNode | undefined;
 			// if fieldSchema is a LeafnodeSchema, we can check that it's a valid type and set the field.
 			if (isPrimitive(modification)) {
-				// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access
-				(node as any)[treeEdit.field] = modification;
+				try {
+					// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access
+					(node as any)[treeEdit.field] = modification;
+				} catch (error) {
+					if (error instanceof UsageError === false) {
+						throw error;
+					}
```

Comment:
Can we replace the `instanceof` with a different kind of check? This would be at risk of not doing what it's supposed to if an app ends up with non-matching versions of Fluid loaded for some reason.

Reply:
Updated in commit e3117aec2f7fcf5d4edf29a075e7e4ad730133ef